### ISSUE TITLE
@microsoft/react-native needs to be published publicly

### ISSUE THREAD

**Issue by matthargett** [Microsoft/react-native-windows#2094]
The current RNWCPP solution, even the FollyWin32 project, doesn't build at all because it references files in node_modules that don't get installed. The vsts-auth step doesn't help, it looks like the package either isn't published in public npm, or is marked as private in public npm.

**Comment by harinikmsft** [comment]
Thanks for filing the issue, @matthargett. We are still working on removing the internal dependencies. You can see there is a note in the readme under [RNWCPP here](https://github.com/Microsoft/react-native-windows/tree/rnwcpp-preview/RNWCPP#note---only-currently-buildable-using-internal-creds) referencing this issue.

@acoates-ms - can you close this issue once you have completed the work?

**Comment by matthargett** [comment]
I did see the note, but didn't know if it was published as a private package on the public npm and therefore not noticed by MS folks doing the testing.

Is the fork of folly being published in that unavailable package somewhere on public github?

**Comment by rozele** [comment]
We should probably create separate issues to figure out fork management for shared code dependencies: ReactCommon C++ from facebook/react-native, Folly, Yoga, and RNTester. We'll likely need to maintain a fork for each as we'll want to iterate without depending on upstream PR process.

**Comment by matthargett** [comment]
It sounds like there are forks internally already, and I was able to make progress by manually copying my own folly branch into place. Can you start with just publishing the package that contains the relevant code so at least the leaf node projects in the RNWCPP solution can build OOTB?

**Comment by acoates-ms** [comment]
Yes we are going to publish the @microsoft/react-native package, which will include the folly fork.  Hopefully within the next few days.  I was actually waiting on pinging you about this fork until I had gotten that done.  But it looks like you found things sooner than that =).

**Comment by rozele** [comment]
Rather than publishing a scoped package, we're going to publish a .tgz version of the package so that we can continue to leverage existing CLI tooling for react-native. This is similar to the approach that Expo takes. @acoates-ms is currently working on this. Given that the fork of react-native is temporary, it seems like this is the best approach.

**Comment by acoates-ms** [comment]
@matthargett , this is somewhat better now.  Depending how motivated you are you could probably get things building today.  The NPM package we use for react-native is now based on https://github.com/Microsoft/react-native.  So you can modify the version dependency in package.json to point to the tar from that tag in the MS/react-native repo.

So if the dependency in RNW/package.json is "react-native@0.57.0-microsoft.6" you can make the dependency point to https://github.com/Microsoft/react-native/archive/v0.57.0-microsoft.6.tar.gz .  Or you could publish that version to your local feed.

We are still working on removing all the internal nuget dependencies.  I dont think you actually need any of them for the UWP code.  -- or at least that the code that is currently using them can be pretty easily deleted.  A bunch of the projects in the solution are for our win32 implementation, which needs more dependencies, including some of these nugets.  But you can probably just disable those projects.

-- Yes we still obviously have work to get this stuff more clean.

**Comment by harinikmsft** [comment]
Main issue tracking this is now - #2079 to remove the link. Closing this as it is no longer needed based on above comment.
